src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -456,97 +444,105 @@ private static void PerformInjection()
             return (Func<object[], T>)setterMethod.CreateDelegate(typeof(Func<object[], T>));
         }
 
-        /// <summary>
-        /// Initializes all the reflection objects it will ever need. Reflection is costly, but it's better to take
-        /// this one time performance hit than to get it multiple times later, or do it lazily and have to worry about
-        /// threading issues. If Initialize has been called before, it will not doing anything.
-        /// </summary>
-        private void Initialize()
-        {
-            lock (this)
-            {
-                if (!this.initialized)
-                {
-                    try
-                    {
-                        // This flag makes sure we only do this once. Even if we failed to initialize in an
-                        // earlier time, we should not retry because this initialization is not cheap and
-                        // the likelihood it will succeed the second time is very small.
-                        this.initialized = true;
-
-                        PrepareReflectionObjects();
-                        PerformInjection();
-                    }
-                    catch (Exception ex)
-                    {
-                        // If anything went wrong, just no-op. Write an event so at least we can find out.
-                        this.Write(InitializationFailed, new { Exception = ex });
-                    }
-                }
-            }
-        }
-
         private void RaiseRequestEvent(HttpWebRequest request)
         {
-            if (this.IsRequestInstrumented(request))
+            if (!ActivitySource.HasListeners() || this.IsRequestInstrumented(request))
             {
-                // This request was instrumented by previous RaiseRequestEvent, such is the case with redirect responses where the same request is sent again.
+                // No subscribers to the ActivitySource or this request was instrumented by previous
+                // RaiseRequestEvent, such is the case with redirect responses where the same request is sent again.
                 return;
             }
 
-            if (this.IsEnabled(ActivityName, request))
-            {
-                // We don't call StartActivity here because it will fire into user code before the headers are added.
+            var activity = ActivitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                Activity.Current?.Context ?? default,
+                this.BuildRequestTags(request));
 
-                var activity = new Activity(ActivityName);
-                activity.Start();
+            Debug.Assert(activity != null, "Activity null check failed.");
```

Comment:
activity being null is valid here. (sampler off => no activity created, just like when no listener is present)

Reply:
Makes sense, I will fix.